coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/DataTypeDerivedSqlCallConverter.java

```diff
@@ -44,7 +45,7 @@ public DataTypeDerivedSqlCallConverter(HiveMetastoreClient mscClient, SqlNode to
     operatorTransformerList = SqlCallTransformers.of(new FromUtcTimestampOperatorTransformer(typeDerivationUtil),
         new GenericProjectTransformer(typeDerivationUtil), new NamedStructToCastTransformer(typeDerivationUtil),
         new ConcatOperatorTransformer(typeDerivationUtil), new SubstrOperatorTransformer(typeDerivationUtil),
-        new UnionSqlCallTransformer(typeDerivationUtil));
+        new CastOperatorTransformer(typeDerivationUtil), new UnionSqlCallTransformer(typeDerivationUtil));
```

Comment:
is the ordering relevant here? Generally we add the new transformer at the end. But in this case, `CastOperatorTransformer` is appended before `UnionSqlCallTransformer`

Reply:
Order shouldn't matter here since `UnionSqlCallTransformer` doesn't create any types of cast that `CastOperatorTransformer` would need to transform.